src/lib/running-list/components/RunningList.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+
+function RunningList({ children, className }) {
+  const [animationWidth, setAnimationWidth] = useState(0);
+
+  const [mounted, setMounted] = useState(false);
+
+  const listContainerRef = useRef();
+
+  useEffect(() => {
+    setMounted(true);
+
+    return () => setMounted(false);
+  }, [setMounted]);
+
+  useEffect(() => {
+    function updateWidth() {
+      setTimeout(() => {
+        if (listContainerRef && listContainerRef.current) {
+          setAnimationWidth(listContainerRef.current.offsetWidth);
+        }
+      }, 100);
+    }
+    if (listContainerRef && listContainerRef.current) {
+      window.addEventListener('resize', updateWidth);
+      listContainerRef.current.addEventListener('animationstart', updateWidth);
```

Comment:
Is this completely controlled by css? Maybe it has to import the css file itself, then? 😅 Not sure what makes most sense tbh

Reply:
True that! It is completely controlled by css - probably makes the most sense that this itself imports the css file 😅